sdk/platform/http_client/curl/inc/curl_client.hpp

```diff
@@ -1,41 +1,86 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) `Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
 #pragma once
 
 #include <curl/curl.h>
 #include <http/http.hpp>
+#include <memory>
 
 class CurlClient {
 private:
+  CurlClient(const CurlClient&) = delete;
+  CurlClient& operator=(const CurlClient&) = delete;
+
   Azure::Core::Http::Request& m_request;
```

Comment:
This reference member fails the const-or-reference-member guideline but more importantly suggests that this is really `CurlRequest` rather than `CurlClient` since it may only ever be used with one request (ever).